tests/integration/test_schema.py

```diff
@@ -1079,85 +1082,159 @@ async def assert_schema_versions_failed(client: Client, trail: str, schema_id: i
     assert res.status_code == response_code
 
 
-async def register_schema(registry_async_client: Client, trail, subject: str, schema_str: str) -> Tuple[int, int]:
+async def register_schema(
+    registry_async_client: Client, trail: str, subject: str, schema_str: str, schema_type: SchemaType = SchemaType.AVRO
+) -> Tuple[int, int]:
     # Register to get the id
+    payload = {"schema": schema_str}
+    if schema_type == SchemaType.JSONSCHEMA:
+        payload["schemaType"] = "JSON"
```

Comment:
**[non-blocking]**
we could use the enum values to set `payload["schemaType"]`

Reply:
Fixed, good catch.